packages/trip-details/src/index.tsx

```diff
@@ -0,0 +1,257 @@
+import coreUtils from "@opentripplanner/core-utils";
+import moment from "moment";
+import React, { ReactElement } from "react";
+import { FormattedMessage, FormattedNumber } from "react-intl";
+import { CalendarAlt, Heartbeat, MoneyBillAlt } from "styled-icons/fa-solid";
+
+import * as Styled from "./styled";
+import TripDetail from "./trip-detail";
+import { CaloriesDetailsProps, TripDetailsProps } from "./types";
+
+// Load the default messages.
+import defaultEnglishMessages from "../i18n/en-US.yml";
+
+interface TripDetailsMessages {
+  calories: string;
+  caloriesDescription: string;
+  departure: string;
+  title: string;
+  tncFare: string;
+  transitFare: string;
+}
+
+// Extract the TripDetails portion of the default message structure.
+const defaultMessages = defaultEnglishMessages.otpUi.TripDetails;
+
+/**
+ * Construct a map to get message ids (use keys from defaultMessages).
+ */
+// TODO: Find a better place for this utility.
+function getMessageIds<T>(sourceObject: T, idPrefix: string): T {
+  const ids = {};
+  Object.keys(sourceObject).forEach((k: string) => {
+    ids[k] = `${idPrefix}.${k}`;
+  });
+  return ids;
+}
+
+const messageIds = getMessageIds<TripDetailsMessages>(
+  defaultMessages,
+  "otpUi.TripDetails"
+);
```

Comment:
I'm kind of wondering if all of this is really necessary. It is possible to simply provide string literals for the message IDs and default messages in each `FormattedMessage` call. I suppose this approach might help with static typing to make sure the messages are coded correctly, but there is still no guarantee that the imported data from the messages yml file will have exact matches (especially with the default messages which could prove to be a problem if that fallback were to occur). There does exist some helpful [tooling](https://formatjs.io/docs/tooling/cli) that can automatically detect message definitions and also the default messages which could then be compared against what exists in the yml files. Additionally, removing this might simplify things a bit for downstream implementers so that they aren't required to input yml files in their transpilation workflow.

Reply:
I felt it was necessary to collect message ids because each id is used twice in each `FormattedMessage`, once to set the id of that message, and once to set the default English message.

Reply:
Yeah, but the point still stands that if the messages aren't actually defined in the yml file, then it won't work properly. Furthermore, now that I think about it, this is probably incompatible with how the formatjs cli extracts message IDs (see [here](https://formatjs.io/docs/tooling/cli)).

Reply:
Partially addressed by https://github.com/opentripplanner/otp-ui/pull/281/commits/9725b4bf3d43303d383b1d42219fe199c3b7601e, bu I suspect we should use the `defineMessages` API?

Reply:
`defineMessages` API added in https://github.com/opentripplanner/otp-ui/pull/281/commits/298ffc9e50465de75a7f4c365fc3f12e6db9af16.